src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisition.cs

```diff
@@ -310,6 +325,47 @@ public async Task<AuthenticationResult> GetAuthenticationResultForUserAsync(
             }
         }
 
+        private async Task<AuthenticationResult?> GetAuthenticationResultForConfidentialClientUsingRopcAsync(IConfidentialClientApplication application, IEnumerable<string> scopes, ClaimsPrincipal? user, MergedOptions mergedOptions, TokenAcquisitionOptions? tokenAcquisitionOptions)
+        {
+            if (user != null && user.HasClaim(c => c.Type == ClaimConstants.Username) && user.HasClaim(c => c.Type == ClaimConstants.Password))
+            {
+                string username = user.FindFirst(ClaimConstants.Username)?.Value ?? string.Empty;
+                string password = user.FindFirst(ClaimConstants.Password)?.Value ?? string.Empty;
+
+                var accounts = await application.GetAccountsAsync().ConfigureAwait(false);
```

Comment:
I don't think that this exists in confidential client applications? does it?
or does it only contain the ROPC accounts?

We need to be very careful here.

Reply:
It exists in confidential client applications. It is currently obsolete but we still use in MSAL tests. I have created this [PR](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/4976) in MSAL.Net to remove the obsolete attribute from this method as suggested by @bgavrilMS

Reply:
Wow. I need explanations here.
- which account will they return be in a web app that signs-in users?

Reply:
Agree that this seems to be going backwards.

Reply:
It exists and it is used @jmprieur, despite the API being deprecated. Some apps who choose to integrate directly with MSAL.NET for web app sceanrio will serialize their cache (for a single user) to the session. Then they apply the `GetAccounts` pattern. There is no need to remember the account_id in the session, it just extra info.

So in hindsight, the api makes sense if you know what you are doing. We kept it obsolete because we want ppl to use Id.Web and not MSAL directly and Id.Web holds on to the extra information in ClaimsPrincipal. 

`which account will they return be in a web app that signs-in users` - whatever account(s) you serialized. 

We can add API comments that it should not be used in web api scenario, but only with ROPC.

Reply:
Alternative: Add support for `GetAccountById(null)` which will return the single cached account. And throw an exception if there are 2+ cached accounts.

Thoughts?

Reply:
@jmprieur , @bgavrilMS Added an alternative to GetAccountsAsync by adding a claim account id to the user after first token acquisition. Please review.